app/modules/net_info.c

```diff
@@ -0,0 +1,197 @@
+/* 
+ * http://blog.mclemon.io/esp8266-contributing-to-the-nodemcu-ecosystem
+ * https://github.com/smcl/nodemcu-firmware/blob/cc04aaf92c1c076c30ef0b0eee43b3f924137440/app/modules/test.c
+ *
+ * extracted ping function
+ * adopted jan 2017 for commit to nodeMCU by Wolfgang Rosner
+ */
+
+
+#include "module.h"
+#include "lauxlib.h"
+#include "platform.h"
+
+// #include <string.h>
+// #include <strings.h>
+// #include <stddef.h>
+// #include <stdint.h>
+#include "mem.h"
+
+#include "lwip/ip_addr.h"
+#include "espconn.h"
+#include "lwip/dns.h" 
+#include "lwip/app/ping.h"
+#include "lwip/raw.h"
+
+
+#include "task/task.h"
+
+
+
+// https://github.com/nodemcu/nodemcu-firmware/wiki/[DRAFT]-How-to-write-a-C-module#debug-and-error-messages
+
+#define NET_INFO_DEBUG_ON
+
+#define log_prefix " ### DEBUG: net_info ### :  "
+
+#if defined(DEVELOP_VERSION)
+  #define NET_INFO_DEBUG_ON
+#endif
+#if defined(NET_INFO_DEBUG_ON)
+  #define NET_INFO_DEBUG(format, ...) dbg_printf("%s"format"", log_prefix, ##__VA_ARGS__)
+#else
+  #define NET_INFO_DEBUG(...)
+#endif
+#if defined(NODE_ERROR)
+  #define NET_INFO_ERR(format, ...) NODE_ERR("%s"format"\n", log_prefix, ##__VA_ARGS__)
+#else
+  #define NET_INFO_ERR(...)
+#endif
+
+
+#define NET_INFO_PRIORITY_OUTPUT   TASK_PRIORITY_MEDIUM
+#define NET_INFO_PRIORITY_ERROR    TASK_PRIORITY_MEDIUM
+
+// these statics should go away for reentrancy and maybe other reasons!
+static int ping_callback_ref; 
```

Comment:
You can use the oddly-named `reverse` field of the `struct ping_option` for this, rather than it being global, so that there can be multiple pings outstanding at once.